plume-models/src/posts.rs

```diff
@@ -55,6 +57,54 @@ pub struct NewPost {
     pub source: String,
 }
 
+impl Provider<PgConnection> for Post {
+    type Data = PostEndpoint;
+
+    fn get(conn: &PgConnection, id: i32) -> Result<PostEndpoint, Error> {
+        Post::get(conn, id).map(|p| Ok(PostEndpoint {
+            id: Some(p.id),
+            title: Some(p.title.clone()),
+            subtitle: Some(p.subtitle.clone()),
+            content: Some(p.content.get().clone())
+        })).unwrap_or(Err(Error::NotFound("Get Post".to_string())))
+    }
+
+    fn list(conn: &PgConnection, filter: PostEndpoint) -> Vec<PostEndpoint> {
+        let mut query = posts::table.into_boxed();
+        if let Some(title) = filter.title {
+            query = query.filter(posts::title.eq(title));
+        }
+        if let Some(subtitle) = filter.subtitle {
+            query = query.filter(posts::subtitle.eq(subtitle));
+        }
+        if let Some(content) = filter.content {
+            query = query.filter(posts::content.eq(content));
+        }
```

Comment:
what's the point of these queries and the override of the query variable?

Reply:
aahh, you're fetching different parts of the same article, and stuffing them into different boxes of the query variable!

Reply:
actually, the `PostEndpoint` type is here to represent both the client's query and the result sent by the API. `filter` is what is requested by the client, so if any filter have been specified I update the SQL `query` to only return the requested posts.

Reply:
not sure if I'm clear, tell me if not :grin: